src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/SymmetricAlgorithm.cs

```diff
@@ -428,6 +428,316 @@ public int GetCiphertextLengthCfb(int plaintextLength, PaddingMode paddingMode =
+        /// <param name="paddingMode">The padding mode used to produce the ciphertext and remove during decryption.</param>
+        /// <returns>The decrypted plaintext data.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="ciphertext" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="paddingMode" /> is not a valid padding mode.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <see cref="TryDecryptEcbCore" /> could not decrypt the ciphertext.
+        /// </exception>
+        /// <remarks>
+        ///   This method's behavior is defined by <see cref="TryDecryptEcbCore" />.
+        /// </remarks>
+        public byte[] DecryptEcb(byte[] ciphertext, PaddingMode paddingMode)
+        {
+            // Padding mode is validated by callee.
+            if (ciphertext is null)
+                throw new ArgumentNullException(nameof(ciphertext));
+
+            return DecryptEcb(new ReadOnlySpan<byte>(ciphertext), paddingMode);
+        }
+
+        /// <summary>
+        ///   Decrypts data using ECB mode with the specified padding mode.
+        /// </summary>
+        /// <param name="ciphertext">The data to decrypt.</param>
+        /// <param name="paddingMode">The padding mode used to produce the ciphertext and remove during decryption.</param>
+        /// <returns>The decrypted plaintext data.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="paddingMode" /> is not a valid padding mode.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <see cref="TryDecryptEcbCore" /> could not decrypt the ciphertext.
+        /// </exception>
+        /// <remarks>
+        ///   This method's behavior is defined by <see cref="TryDecryptEcbCore" />.
+        /// </remarks>
+        public byte[] DecryptEcb(ReadOnlySpan<byte> ciphertext, PaddingMode paddingMode)
+        {
+            CheckPaddingMode(paddingMode);
+
+            // This could get returned directly to the caller if we there was no padding
+            // that needed to get removed, so don't rent from a pool.
+            byte[] decryptBuffer = GC.AllocateUninitializedArray<byte>(ciphertext.Length);
+
+            if (!TryDecryptEcbCore(ciphertext, decryptBuffer, paddingMode, out int written)
+                || (uint)written > decryptBuffer.Length)
+            {
+                // This means decrypting the ciphertext grew in to a larger plaintext or overflowed.
+                // A user-derived class could do this, but it is not expected in any of the
+                // implementations that we ship.
+
+                throw new CryptographicException(SR.Argument_DestinationTooShort);
+            }
+
+            if (written == decryptBuffer.Length)
```

Comment:
Nit: I'd replace this entire _if / else if / else_ block with:
```cs
Array.Resize(ref decryptBuffer, written);
return decryptBuffer;
```

Let `Array.Resize` do the optimal thing. Shouldn't need to optimize for the zero-length array case given its rarity.

(Same comment in similar code patterns.)

Reply:
Huh. I didn't realize `Array.Resize` could be used to shrink an array. I've only ever used it for "growing".